### PR TITLE
Deploy b network

### LINKED ISSUE #25: Deploy Upgradeable Smart Contract in Besu Network
Deploy Upgradeable Smart Contract in Besu Network.

Modify `trufle-config.json` if needed.

### PR DESCRIPTION
**Pull request by alejandroalffer**
Updated files to be used fot deploy SmartContracts in AlastriaB network

### CODE REVIEW

File: _hidden-secret [hunk 1]
```diff
@@ -0,0 +1 @@
+avoid hole fish girl immune mammal link fiscal steak pigeon oppose path purpose member market
```
Comment (VictorNS69) [inline _hidden-secret:1]: Remove this file as we already have the `.secret`.

File: truffle-config-b.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
+/**
```
Comment (VictorNS69) [inline truffle-config-b.js:1]: Why do we have this file, if we can use the already existing `truffle-config.js` file?

File: addresses-b.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"Eidas":"0xb8bCaFdEb78Ae5Aa686c5D00616e1561Db0309d4","AlastriaServiceProvider":"0x83D65fAc900dB0A27B45854AC64DA3D25d97864d","AlastriaIdentityIssuer":"0xE8Dc811f7D05c44EfDD998FFfC26D610B0E5471e","AlastriaCredentialRegistry":"0x2f384716e73ae165bC3427Fd60550c26e6CB91c6","AlastriaPresentationRegistry":"0x95bC1119b764Bd678B68A1c03Cd8c06EfA27Df81","AlastriaPublicKeyRegistry":"0x19e33b2424c1D7B5E563E0A74d8b10CE3812469C","AlastriaIdentityManager":"0xE9B9a6E7700672DBbCe59664457D7E6F1dc00F3c","AlastriaNameService":"0xC1b1Fc2434D50966c28C8573AbdF3c7842a2Abb4"}
```
Comment (VictorNS69) [inline addresses-b.json:1]: Indent this file to make it easier to read.

**Review by VictorNS69 — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -1,35 +1,65 @@
-# WIP  temp-alastriaID-truffle-contracts ---
```
Comment (nefera606) [inline README.md:1]: Can we add section about how to use the scripts in the package and how to deploy contracts in a local env?

File: addresses-b.json (hunk 3 above)
Comment (nefera606) [reply addresses-b.json:1]: I think it is generated from fs.write, maybe we need to update the write format

File: migrations/4_initialize_contracts.js [hunk 5]
```diff
@@ -66,5 +66,5 @@ module.exports = async function (callback) {
   }
 
   console.log('Contracts initialized');
-  callback(null);
+  // callback(null);
```
Comment (nefera606) [inline migrations/4_initialize_contracts.js:69]: Can we leave a comment here to say that this null is needed and why? I can remember is part of truffle, but to be sure.

File: package.json [hunk 6]
```diff
@@ -33,7 +33,7 @@
   "license": "MIT",
   "dependencies": {
     "@truffle/config": "1.2.33",
-    "@truffle/hdwallet-provider": "1.1.1",
+    "@truffle/hdwallet-provider": "^1.1.1",
```
Comment (nefera606) [inline package.json:36]: Please, remove ^ versions, for easy management of updates, we must provide fixed versions

File: truffle-config-b.js (hunk 2 above)
Comment (nefera606) [reply truffle-config-b.js:1]: completely agree

File: truffle-config.js [hunk 7]
```diff
@@ -82,15 +84,20 @@ module.exports = {
...
     'red-b-first-id': {
       gasPrice: 0x0,
       provider: () => {
         return new HDWalletProvider(firstIdKey, bNetworkNode);
       },
-      network_id: "*",
+      network_id: "2020",
+    },
+    "alastria-id-b": {
+      provider: () => new HDWalletProvider(mnemonic, `http://63.33.206.111:8545`),
```
Comment (nefera606) [inline truffle-config.js:98]: Can we move the IP to something like RED_B_NODE_IP? I mean, could be interesting at some point to move the IP configs to a config file or env, its a changing config that does not depends on the network configuration (each member of Alastria will be using their nodes, with the same chain config)

Comment (nefera606) [inline truffle-config.js:98]: Can we move the IP to something like RED_B_NODE_IP? I mean, could be interesting at some point move the IP configs to a config file or env, its a changing config that does not depends on the network configuration (each member of Alastria will be using their nodes, with the same chain config)

**Review by nefera606 — CHANGES_REQUESTED** [review]

File: migrations/4_initialize_contracts.js (hunk 5 above)
Comment (xdaniortega) [reply migrations/4_initialize_contracts.js:69]: Totally agree on that, I also have it commented, otherwise Truffle throws an error.

**Comment by VictorNS69** [comment]
This PR is closed as it is obsolete. The updated code is in the branch https://github.com/alastria/alastriaID-truffle-contracts/tree/deploy-b-network

**Pull request closed without merging**